addons/account/models/sequence_mixin.py

```diff
@@ -41,6 +41,19 @@ def init(self):
                     field=self._sequence_field,
                 ))
 
+    def write(self, vals):
+        for record in self:
+            date_f = record._sequence_date_field
+            if date_f in vals and record[date_f] != vals[date_f]:
+                # Ensure consistency between date and sequence
+                format_values = record._get_sequence_format_param(record[record._sequence_field] or '')[1]
```

Comment:
```suggestion
                format_values = record._get_sequence_format_param(vals.get(record._sequence_field, record[record._sequence_field]) or '')[1]
```
 if we change the date AND the sequence t the same time?